src/integrations/__tests__/legacy.spec.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import removeLegacySymlinks from '@/integrations/legacy';
+
+const resourcesDir = path.join('resources', os.platform(), 'bin');
+let testDir: string;
+let someOtherDir: string;
+let legacyIntegrationDir: string;
+
+beforeEach(async() => {
+  testDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rdtest-'));
+  someOtherDir = path.join(testDir, 'someOtherDir');
+  await fs.promises.mkdir(someOtherDir);
+  legacyIntegrationDir = path.join(testDir, 'legacyIntegrationDir');
+  await fs.promises.mkdir(legacyIntegrationDir);
+});
+
+afterEach(async() => {
+  // It is best to be careful around rm's; we don't want to remove important things.
+  if (testDir) {
+    await fs.promises.rm(testDir, { recursive: true, force: true });
+  }
+});
+
+test('Ensure legacy symlinks are removed properly', async() => {
+  // make managed symlinks (these should be removed)
+  const managedLegacySymlinks = ['docker', 'kubectl'];
+
+  for (const name of managedLegacySymlinks) {
+    const resourcesPath = path.join(resourcesDir, name);
+    const legacyIntegrationPath = path.join(legacyIntegrationDir, name);
+
+    await fs.promises.symlink(resourcesPath, legacyIntegrationPath);
+  }
+
+  // make unmanaged symlinks (these should not be removed)
+  const unmanagedLegacySymlinks = ['helm', 'nerdctl'];
+
+  for (const name of unmanagedLegacySymlinks) {
+    const resourcesPath = path.join(resourcesDir, name);
+    const someOtherPath = path.join(someOtherDir, name);
+    const legacyIntegrationPath = path.join(legacyIntegrationDir, name);
+
+    await fs.promises.symlink(resourcesPath, someOtherPath);
+    await fs.promises.symlink(someOtherPath, legacyIntegrationPath);
+  }
+
+  await removeLegacySymlinks(legacyIntegrationDir);
+
+  const remaining = await fs.promises.readdir(legacyIntegrationDir);
+
+  expect(remaining.length).toEqual(unmanagedLegacySymlinks.length);
```

Comment:
I think you can replace lines 53-56 with `expect(remaining).toContainEqual(unmanagedLegacySymlinks)`

Reply:
I just gave that a shot, because it should work, but it doesn't seem to:
```
/home/adam/dev/suse/rancher-desktop〉npm run test:unit:jest --testPathPattern legacy 03/25/2022 10:41:41 AM

> rancher-desktop@1.1.0 test:unit:jest
> jest "legacy"

 FAIL  src/integrations/__tests__/legacy.spec.ts
  ✕ Ensure legacy symlinks are removed properly (19 ms)

  ● Ensure legacy symlinks are removed properly

    expect(received).toContainEqual(expected) // deep equality

    Expected value: ["helm", "nerdctl"]
    Received array: ["helm", "nerdctl"]

      51 |   const remaining = await fs.promises.readdir(legacyIntegrationDir);
      52 |
    > 53 |   expect(remaining).toContainEqual(unmanagedLegacySymlinks);
         |                     ^
      54 |   //expect(remaining.length).toEqual(unmanagedLegacySymlinks.length);
      55 |   //for (const name of remaining) {
      56 |   //  expect(unmanagedLegacySymlinks).toContain(name);

      at Object.<anonymous> (src/integrations/__tests__/legacy.spec.ts:53:21)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        4.169 s, estimated 6 s
Ran all test suites matching /legacy/i.
```

Reply:
It's puzzling. The docs say that it recursively checks each field for equality rather than identity, so it should work :shrug:

Reply:
After some more messing around, it seems that `expect(remaining).toEqual(unmanagedLegacySymlinks)` does the trick. Even shorter!